frontend/src/main/web/lib/stores/GlossaryStore.js

```diff
@@ -0,0 +1,522 @@
+import Dispatcher from '../dispatchers/GlossaryDispatcher';
+import assign from 'object-assign';
+import {EventEmitter} from 'events';
+import {Promise} from 'es6-promise';
+import Request from 'superagent';
+import Configs from '../constants/Configs';
+import {GlossaryActionTypes} from '../constants/ActionTypes';
+import GlossaryHelper from '../utils/GlossaryHelper'
+import StringUtils from '../utils/StringUtils'
+import DateHelpers from '../utils/DateHelper'
+import _ from 'lodash';
+
+var SIZE_PER_PAGE = 5000, CHANGE_EVENT = "change",
+    MAX_LISTENER = 100; //number of listener for GlossaryStore (default is 11)
+
+EventEmitter.prototype.setMaxListeners(MAX_LISTENER);
+
+var _state = {
+  canAddNewEntry: false,
+  canUpdateEntry: false,
+  localeOptions: [],
+  srcLocale: null,
+  selectedTransLocale: null,
+  locales: null,
+  glossary: {},
+  original_glossary: {},
+  glossaryResId: [],
+  page:1,
+  filter: '',
+  focusedRow: {
+    rowIndex: -1
+  },
+  uploadFile: {
+    show: false,
+    status : -1,
+    file: null,
+    transLocale: null
+  },
+  sort: {
+    src_content: true
+  },
+  totalCount: 0,
+  newEntry: {
+    term: '',
+    pos: '',
+    description: '',
+    isSaving: false,
+    show: false,
+    isAllowSave: false
+  }
+};
+
+function localesStatAPIUrl() {
+  return Configs.baseUrl + "/glossary/info" + Configs.urlPostfix
+}
+
+function loadLocalesStats() {
+  var url = localesStatAPIUrl();
+
+  return new Promise(function(resolve, reject) {
+    Request.get(url)
+      .set("Cache-Control", "no-cache, no-store, must-revalidate")
+      .set('Accept', 'application/json')
+      .set("Pragma", "no-cache")
+      .set("Expires", 0)
+      .end(function (err, res) {
+        if(err != null && err.error === true) {
+          console.error(url, err);
```

Comment:
As long as your code gets what it expects. There could be a better way to structure the page if it freezes waiting for a promise to resolve, but that is probably too much to change at the moment.

Just for future reference, the basic approach that I will use for zanata-spa is have an async action to do the request, and make it dispatch actions depending what stage it is at and what happens. Something like:

```javascript
function someAction (someParam) {
  return (dispatch) => {
    dispatch(fetchingStuff())
    Request.get(someParam)... // etc.
      .end((err, res) => {
        // whatever else to check for errors
        if (err) {
          // could use fetchedStuff() for both error and success and give different params
          // just depends how the actions and action creators are designed
          dispatch(fetchingStuffFailed(err)
          return
        }
        dispatch(fetchedStuff(res.stuff)
      })
  }
}
```